main/cmds.go

```diff
@@ -339,3 +337,22 @@ func decodeScript(script string) (string, string, error) {
 	w.Flush()
 	return buf.String(), fmt.Sprintf("%d;%d;gzip=1", len(script), n), nil
 }
+
+func clearSettingsAndScriptExceptMostRecent(seqNum) {
+	downloadsParent := filepath.Join(dataDir, downloadDir)
+	seqNumString := strconv.Itoa(seqNum)
+
+	ctx.Log("event", "clearing settings and script files except most recent seq num")
+	err = utils.TryDeleteDirectoriesExcept(downloadsParent, seqNumString)
```

Comment:
can TryDeleteDirectoriesExcept and TryClearRegexMatchingFilesExcept return error even if some directories/files are deleted and some aren't? in that case it would help to log which files could not be deleted. Logging a single aggregate message such as "could not clear scripts" is less useful than logging the names of files that could not be deleted. Note that this can be done inside of the Try* methods and you don't necessarily have to return a list of file/directory names to their caller. The caller can simply log "some scripts could not be deleted" or "some settings files could not be deleted"

Reply:
I'll address this in a separate PR in the common repo!